CoreTesting/src/au/gov/asd/tac/constellation/testing/construction/ImageGraphBuilderPlugin.java

```diff
@@ -134,127 +154,141 @@ protected void edit(final GraphWriteMethods graph, final PluginInteraction inter
+
+        final int pixelXAttributeId = graph.addAttribute(GraphElementType.VERTEX, FloatAttributeDescription.ATTRIBUTE_NAME, "pixelX", "pixelX", "", null);
+        final int pixelYAttributeId = graph.addAttribute(GraphElementType.VERTEX, FloatAttributeDescription.ATTRIBUTE_NAME, "pixelY", "pixelY", "", null);
+        final int redAttributeId = graph.addAttribute(GraphElementType.VERTEX, FloatAttributeDescription.ATTRIBUTE_NAME, "red", "red", "", null);
+        final int greenAttributeId = graph.addAttribute(GraphElementType.VERTEX, FloatAttributeDescription.ATTRIBUTE_NAME, "green", "green", "", null);
+        final int blueAttributeId = graph.addAttribute(GraphElementType.VERTEX, FloatAttributeDescription.ATTRIBUTE_NAME, "blue", "blue", "", null);
+        final int alphaAttributeId = graph.addAttribute(GraphElementType.VERTEX, FloatAttributeDescription.ATTRIBUTE_NAME, "alpha", "alpha", "", null);
+        final int diffSouthAttributeId = graph.addAttribute(GraphElementType.VERTEX, FloatAttributeDescription.ATTRIBUTE_NAME, "diffSouth", "diffSouth", "", null);
+        final int diffNorthAttributeId = graph.addAttribute(GraphElementType.VERTEX, FloatAttributeDescription.ATTRIBUTE_NAME, "diffNorth", "diffNorth", "", null);
+        final int diffEastAttributeId = graph.addAttribute(GraphElementType.VERTEX, FloatAttributeDescription.ATTRIBUTE_NAME, "diffEast", "diffEast", "", null);
+        final int diffWestAttributeId = graph.addAttribute(GraphElementType.VERTEX, FloatAttributeDescription.ATTRIBUTE_NAME, "diffWest", "diffWest", "", null);
+        final int transactionWeightAttributeId = AnalyticConcept.TransactionAttribute.WEIGHT.get(graph);
+
+        final boolean useVertexAttributes = pixelXAttributeId != Graph.NOT_FOUND
+                && pixelYAttributeId != Graph.NOT_FOUND
+                && greenAttributeId != Graph.NOT_FOUND
+                && redAttributeId != Graph.NOT_FOUND
+                && blueAttributeId != Graph.NOT_FOUND
+                && alphaAttributeId != Graph.NOT_FOUND
+                && diffSouthAttributeId != Graph.NOT_FOUND
+                && diffNorthAttributeId != Graph.NOT_FOUND
+                && diffEastAttributeId != Graph.NOT_FOUND
+                && diffWestAttributeId != Graph.NOT_FOUND;
+        final boolean useTransAttributes = transactionWeightAttributeId != Graph.NOT_FOUND;
+
+        int layer = 1;
+        int w = 0;
+        int prevWidth = 0;
+        final boolean addRight = (boolean) parameters.getObjectValue(ADD_RIGHT_PARAMETER_ID);
+        final boolean addLayers = (boolean) parameters.getObjectValue(ADD_LAYERS_PARAMETER_ID);
+        final int stateAttributeId = LayersViewConcept.MetaAttribute.LAYERS_VIEW_STATE.ensure(graph);
+        final LayersViewState currentState = graph.getObjectValue(stateAttributeId, 0);
+        if (currentState != null) {
+            // set start layer to existing graph layer count + 1
+            layer = currentState.getLayerCount() + 1;
+        }
```

Comment:
I wonder if this portion of code could only be executed when `addLayers` is true? That way, this attribute is only added and used if there is a need to use it